server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeIssueMeasuresStep.java

```diff
@@ -0,0 +1,341 @@
+  }
+
+  private static double sumChildrenMeasuresOnPeriod(Collection<Measure> measures, int periodIndex) {
+    SumVariationMeasure sumMeasures = new SumVariationMeasure(periodIndex);
+    from(measures).filter(sumMeasures).size();
+    return sumMeasures.getSum();
+  }
+
+  private static int sumIssuesOnPeriod(Collection<Issue> issues, long periodDate) {
+    // Add one second to not take into account issues created during current analysis
+    long datePlusOneSecond = periodDate + 1000L;
+    SumIssueAfterDate sumIssues = new SumIssueAfterDate(datePlusOneSecond);
+    from(issues).filter(sumIssues).toList();
+    return sumIssues.getSum();
+  }
+
+  private static class CountIssues {
+    int openIssues = 0;
+    int reopenedIssues = 0;
+    int confirmedIssues = 0;
+    Multiset<String> severityBag = HashMultiset.create();
+    ListMultimap<String, Issue> issuesPerSeverity = ArrayListMultimap.create();
+
+    public CountIssues(Iterable<Issue> issues) {
+      for (Issue issue : issues) {
+        countByStatus(issue.getStatus());
+        severityBag.add(issue.getSeverity().name());
+        issuesPerSeverity.put(issue.getSeverity().name(), issue);
+      }
+    }
+
+    private void countByStatus(String status) {
+      switch (status) {
+        case STATUS_OPEN:
+          openIssues++;
+          break;
+        case STATUS_REOPENED:
+          reopenedIssues++;
+          break;
+        case STATUS_CONFIRMED:
+          confirmedIssues++;
+          break;
+        default:
+          // Other statuses are ignored
+      }
+    }
+  }
+
+  private static class SumMeasure implements Predicate<Measure> {
+
+    private int sum = 0;
+
+    @Override
+    public boolean apply(@Nonnull Measure input) {
+      sum += input.getIntValue();
+      return true;
+    }
+
+    public int getSum() {
+      return sum;
+    }
+  }
+
+  private static class SumVariationMeasure implements Predicate<Measure> {
+
+    private final int periodIndex;
+    private double sum = 0d;
+
+    public SumVariationMeasure(int periodIndex) {
+      this.periodIndex = periodIndex;
+    }
+
+    @Override
+    public boolean apply(@Nonnull Measure input) {
+      if (input.hasVariations() && input.getVariations().hasVariation(periodIndex)) {
+        sum += input.getVariations().getVariation(periodIndex);
+      }
+      return true;
+    }
+
+    public double getSum() {
+      return sum;
+    }
+  }
+
+  private static class SumIssueAfterDate implements Predicate<Issue> {
+
+    private final long date;
+    private int sum = 0;
+
+    public SumIssueAfterDate(long date) {
+      this.date = date;
+    }
+
+    @Override
+    public boolean apply(@Nonnull Issue issue) {
+      if (isAfter(issue, date)) {
+        sum++;
+      }
+      return true;
+    }
+
+    public int getSum() {
+      return sum;
+    }
+
+    private static boolean isAfter(Issue issue, long date) {
+      // TODO should we truncate the date to the second as it was done in batch ?
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)